lib/rpcc_dpdk/rte_flow_ndn.c

```diff
@@ -910,3 +910,133 @@ rpc_rte_flow_indir_action_conf_destroy(rcf_rpc_server *rpcs,
+{
+    tarpc_rte_flow_action_handle_destroy_in   in = {};
+    tarpc_rte_flow_action_handle_destroy_out  out = {};
+    te_log_buf                               *tlbp;
+
+    in.port_id = port_id;
+    in.handle = (tarpc_rte_flow_action_handle)handle;
+
+    rcf_rpc_call(rpcs, "rte_flow_action_handle_destroy", &in, &out);
+
+    CHECK_RETVAL_VAR_IS_ZERO_OR_NEG_ERRNO(rte_flow_action_handle_destroy,
+                                          out.retval);
+
+    tlbp = te_log_buf_alloc();
+    TAPI_RPC_LOG(rpcs, rte_flow_action_handle_destroy,
+                 "port_id=%" PRIu16 ", handle=" RPC_PTR_FMT,
+                 NEG_ERRNO_FMT "%s", in.port_id, RPC_PTR_VAL(in.handle),
+                 NEG_ERRNO_ARGS(out.retval), (out.retval != 0) ?
+                 tarpc_rte_flow_error2str(tlbp, &out.error) : "");
+    te_log_buf_free(tlbp);
+
+    tarpc_rte_flow_error_copy(error, &out.error);
+
+    RETVAL_ZERO_INT(rte_flow_action_handle_destroy, out.retval);
+}
+
+int
+rpc_rte_flow_action_handle_update(rcf_rpc_server *rpcs, uint16_t port_id,
+                                  rpc_rte_flow_action_handle_p handle,
+                                  rpc_rte_flow_action_handle_update_p update,
+                                  tarpc_rte_flow_error *error)
+{
+    tarpc_rte_flow_action_handle_update_in   in = {};
+    tarpc_rte_flow_action_handle_update_out  out = {};
+    te_log_buf                              *tlbp;
+
+    in.port_id = port_id;
+    in.handle = (tarpc_rte_flow_action_handle)handle;
+    in.update = (tarpc_rte_flow_action_handle_update)update;
+
+    rcf_rpc_call(rpcs, "rte_flow_action_handle_update", &in, &out);
+
+    CHECK_RETVAL_VAR_IS_ZERO_OR_NEG_ERRNO(rte_flow_action_handle_update, out.retval);
+
+    tlbp = te_log_buf_alloc();
+    TAPI_RPC_LOG(rpcs, rte_flow_action_handle_update,
+                 "port_id=%" PRIu16 ", handle=" RPC_PTR_FMT ", "
+                 "update=" RPC_PTR_FMT, NEG_ERRNO_FMT "%s",
+                 in.port_id, RPC_PTR_VAL(in.handle), RPC_PTR_VAL(in.update),
+                 NEG_ERRNO_ARGS(out.retval), (out.error.type != 0) ?
+                 tarpc_rte_flow_error2str(tlbp, &out.error) : "");
+    te_log_buf_free(tlbp);
+
+    tarpc_rte_flow_error_copy(error, &out.error);
+
+    RETVAL_ZERO_INT(rte_flow_action_handle_update, out.retval);
+}
+
+int
+rpc_rte_flow_action_handle_query(rcf_rpc_server *rpcs, uint16_t port_id,
+                                 rpc_rte_flow_action_handle_p handle,
+                                 tarpc_rte_flow_query_data *data,
+                                 tarpc_rte_flow_error *error)
+{
+    tarpc_rte_flow_action_handle_query_in   in = {};
+    tarpc_rte_flow_action_handle_query_out  out = {};
+    te_log_buf                             *tlbp_err;
+    te_log_buf                             *tlbp_data;
+
+    in.port_id = port_id;
+    in.handle = (tarpc_rte_flow_action_handle)handle;
+    in.data = *data;
```

Comment:
It should be possible to pass data==NULL

Reply:
1. Yes. Any kind of RPCs should ideally allow bad value testing as well.
2. Ideally, yes
3. It is out-of-scope of the PR.